numpyro/distributions/iaf.py

```diff
@@ -0,0 +1,84 @@
+# adapted from https://github.com/pyro-ppl/pyro/blob/dev/pyro/distributions/transforms/iaf.py
+
+from __future__ import absolute_import, division, print_function
+
+from numpyro.distributions.constraints import Transform, real
+from jax.lax import stop_gradient
+import jax.numpy as np
+
+
+def clamp_preserve_gradients(x, min, max):
+    return x + stop_gradient(np.clip(x, a_min=min, a_max=max) - x)
+
+
+class InverseAutoRegressiveTransform(Transform):
```

Comment:
do i need any `event_dim` logic?

Reply:
I don't think that it is necessary unless you want to use ComposeTransform. It is safe to define `event_dim=0/1` if your `domain` and `codomain` have the same dimension. In the other cases (e.g. domain is 1D, codomain is 2D), the current logic of ComposeTransform logdet will be wrong (something like `coevent_dim` can remedy it). Anyway, you can just left a TODO mark so we can look at it again later. :)

Reply:
ok